data/deprecated.json

```diff
@@ -1166,6 +1438,278 @@
     "old": {"museum_type": "public"},
     "replace": {"operator:type": "public"}
   },
+  {
+    "old": {"name:nan-HJ": "*"},
+    "replace": {"name:nan-Hant": "$1"}
+  },
+  {
+    "old": {"name:nan-POJ": "*"},
+    "replace": {"name:nan-Latn-pehoeji": "$1"}
+  },
+  {
+    "old": {"name:nan-TL": "*"},
+    "replace": {"name:nan-Latn-tailo": "$1"}
+  },
```

Comment:
Some of these deprecations replace very common keys with much less common keys. For example, [`name:nan-TL=*`](https://taginfo.openstreetmap.org/keys/name%3Anan-TL#overview) still has some 38,000 occurrences while [`name:nan-Latn-tailo=*`](https://taginfo.openstreetmap.org/keys/name%3Anan-Latn-tailo) has only eight. I realize that’s part of the motivation for this change, and conforming to the standard is good for interoperability, but have you given a heads-up to the relevant communities yet? Not everyone uses id-tagging-schema–powered editors, and some data consumers may need to update their special cases for nonconformant codes. Given the number of languages involved, consider posting something to the global forum and then reaching out to individual local or language communities separately (such as the Asia Telegram chat).

Reply:
This had been already discussed in both OpenStreetMap Mainland China and OpenStreetMap Taiwan Telegram groups.

- https://t.me/osmchina/93467/107453
- https://t.me/OSMTaiwan/17888

Reply:
Thanks for these links. I can’t read Chinese, but assuming the changes were uncontroversial, the next step would be to alert the broader community. Many software developers don’t participate in the Chinese-language communication channels. Off the top of my head, here are some data consumers that will be affected by the deprecation of `name:zh_pinyin=*` in favor of `name:zh-Latn-pinyin=*` or the deprecation of `name:zh-CN=*`, `name:zh-HK=*`, and `name:zh-TW=*` in favor of `name:zh-Hans-CN=*`, `name:zh-Hant-HK=*`, and `name:zh-Hant-TW=*`, respectively:

* [OsmAnd](https://github.com/osmandapp/OsmAnd-tools/blob/1b6bae3c84482a49f025ee7028c20122426b0450/java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexCreationContext.java#L115)
* [Organic Maps](https://github.com/organicmaps/organicmaps/blob/9c07d29b84c61a81662bb879bff9f49fd3c6ff55/indexer/feature_utils.cpp#L101)
* [OSMUS Tile Service](https://github.com/osmus/tileservice/blob/f2286759341dc81239dde6bc726e93e5d0df270c/renderer/render_once.sh#L51) (used by OSM Americana, OpenTrailMap, etc.)

Additionally, some closed-source products such as [Mapbox Streets](https://docs.mapbox.com/data/tilesets/reference/mapbox-streets-v8/#names) might be affected, but I don’t have any specific information about that.

Please consider starting a conversation somewhere more visible, such as the [tagging forum category](https://community.openstreetmap.org/c/general/tagging/70), and opening issues against the relevant repositories. We don’t have to wait to merge these changes until the data consumers are fully updated, but we should start the discussion ahead of time, as a courtesy. Thanks for your attention to detail!

Reply:
I've asked other community members to handle something about the tagging forum category, waiting for their reply.

We're also checking which repositories are relevant to this change.

Reply:
(2024-10-09 17:55 (UTC) Note: https://github.com/osmlab/name-suggestion-index had been identified.)

Reply:
https://github.com/osmlab/name-suggestion-index/pull/10052